tools/ci_build/github/azure-pipelines/templates/use-android-emulator.yml

```
@@ -13,7 +13,24 @@ parameters:
   type: boolean
   default: false
 
+- name: install_missing_packages
+  type: boolean
+  default: false
+
 steps:
+- ${{ if eq(parameters.install_missing_packages, true) }}:
+  - script: |
+      set -e -x
+      ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --install "emulator"
+      ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --install "platform-tools"
+      ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --list
+      ls -R ${ANDROID_SDK_ROOT}/platform-tools
+      # Do not output ##vso[] commands with `set -x` or they may be parsed again and include a trailing quote.
+      set -x
```

sorry, I made a typo in my original comment's suggestion. it has been edited.
```suggestion
      set +x
```